IntegratedBreedingPlatform/APITests
# Post studies

**List of changes:**
1. Added the ff. tests for POST /v2/studies:

- Verify response code and body when entered complete set of valid inputs for an existing study with no existing design
- Verify response code and body when entered details for multiple instances
- Verify response code and body when entered characters greater than max length of referenceResource field
- Verify response code and body when no trialDbId specified
- Verify response code and body when entered non-existing locationDbId
- Verify response code and body when entered locationDbId that is specific only to other program
- Verify ing details with Crop_season_Code on environment parameters and no seasons parameter
- Verify ing details for an existing study with existing design
- Verify response code and body when entered required details only
- Verify response code and body when entered details for multiple studies
- Verify response code and body when entered invalid value for environment parameters and seasons

2. Added createdTrialDbId to environment variables
3. Update db backup

Tasks | /
-- | --
**Pre-merging** |  
Followed basic standards | ✅  
Ensured that the List of API Services spreadsheet is updated | ✅ 
Ensured that all test data created and used in the tests are added in the Setting up Initial Data (Data Seeding) page | ✅
Ensured that the entire test suite is passing in branch using PostmanCollections-CIRun or PostmanCollections-PerfTests (if changes are related to performance test) jenkins jobs |✅ http://ci.leafnode.io/job/PostmanCollections-CIRun/902/
Ensured that TM4J test cases are updated | ✅  
Created backup of the current test data | ✅  
**Post-merging** |  
Ensured that the entire test suite is passing in master using PostmanCollections-CIRun and PostmanCollections-PerfTests (if changes are related to performance test) jenkins jobs |  
Deleted branch |
